test/BotTest.js

```diff
@@ -618,4 +618,52 @@ describe('Bot', function () {
             expect(classifier).toBe(fakeClassifier);
         });
     });
-});
\ No newline at end of file
+
+    describe('ChainableMethods', function () {
+        it('Chainable methods with train and addSkill', function (done) {
+            var mockClassifier = mockClassifierWithMockClassifierFactory();
+
+            var bot = new Bot({classifierPreference: 'naive_bayes'});
+
+            const fakeSkillFn = function (context, req, res, next) {};
+            const fakeSkill = new Skill('fakeskill', 'topic', fakeSkillFn);
+
+            const skills = bot.train('topic', 'text').addSkill(fakeSkill).getSkills();
```

Comment:
I see what you are trying to do here, however, I think this here is more of an integration test than a unit test. A unit test in this case would be something like:
```javascript
const bot = new Bot();
const returnReference = bot.addSkill(new Skill(...));
expect(returnReference).toBe(bot);
```

You'd have multiple tests for each of the methods that you have changed, testing multiple flows, ensuring that you get the reference back via the right flows.

Unit test needs to be small and concise; something which tests precisely one unit.

What do you think?